### ISSUE TITLE
Unpopulated zone field for Kress RTK

### ISSUE THREAD

**Issue by smornierHA** [MTrab/landroid_cloud#521]
### Describe the issue

Hello, 
First of all thank you very much for you work !
I just wanted to raise an issue regarding the zone field which is not populated for Kress RTK; the value is always 0.
Thanks

### What version of Home Assistant Core has the issue?

core-2024.3.0

### What version of the Landroid Cloud integration do you have installed

4.0.2

### What type of installation are you running?

Home Assistant OS

### Which make and model is the mower used for this integration?

Kress RTK KR174E

### Diagnostics information (NOT log entries!)

none

### Relevant log entries

```txt
none
```

**Comment by MTrab** [comment]
I'd need some credentials sent to landroid_cloud (at) trab.dk to debug on this

**Comment by MTrab** [comment]
This could be fixed by #544 when merged and published

**Comment by MTrab** [comment]
Please update to v4.0.4 and check if this fixes this issue

**Comment by smornierHA** [comment]
Hello,
Thank you for your feedback, unfortunately it's still the same : 
device_class: landroid_cloud__state
friendly_name: RTK
supported_features: 7
accessories:
  cellular: true
  electric_height_adjustment: true
  ultrasonic: true
party_mode_enabled: false
torque: 0
zone:
  current: 0
  index: 0
  indicies:
    - 0
    - 0
    - 0
    - 0
    - 0
    - 0
    - 0
    - 0
    - 0
    - 0
  starting_point:
    - 0
    - 0
    - 0
    - 0
supported_landroid_features: 3903

**Comment by MTrab** [comment]
Then there is no way around me getting access to a device like this, for trying to fix it

**Comment by johngianni** [comment]
@MTrab would this yaml be sufficient? I've tried to fix this as well, finding a way to force the Zone Number into the correct variable. But after the last release things have changed quite a bit, so I give up. AS you can see, the zone number is in a completely different place , peculiar to the RTK protocol. I would give you credentials, but don't know where to send a direct message. 

`{
  "cfg":{
    "id":0,
    "lg":"de","tz":"Europe/Rome",
    "rd":0,
    "cmd":0,
    "tq":0,
    "al":{"lvl":0,"t":60},"cut":{"t":1,"z":[9,28],"d":-1,"rm":0},
    "sc":{"enabled":1,"paused":0,"freq":0,"slots":[{"d":1,"e":1,"s":0,"t":15,
  "cfg":{"cut":{"z":[29]}}},{"d":5,"e":1,"s":0,"t":780,
  "cfg":{"cut":{"z":[25]}}}]},
    "modules":{"US":{"enabled":1},
      "4G":{"enabled":0,"geo":{"dist":50}},
      "EA":{"h":45}},
    "rtk":{
      "map":"x",
      "ck":"x","st":1,"zs":[{
    "id":2,
  "cfg":{
    "modules":{
      "EA":{"h":45}},
    "sc":{"freq":2880},"cut":{"t":1,"d":-1}}},{
    "id":8,
  "cfg":{
    "modules":{
      "EA":{"h":55}},
    "sc":{"freq":2880},"cut":{"t":1,"d":-1}}},{
    "id":9,
  "cfg":{
    "modules":{
      "EA":{"h":55}},
    "sc":{"freq":2880},"cut":{"t":1,"d":-1}}},{
    "id":13,
  "cfg":{
    "modules":{
      "EA":{"h":55}},
    "sc":{"freq":2880},"cut":{"t":1,"d":-1}}},{
    "id":16,
  "cfg":{
    "modules":{
      "EA":{"h":55}},
    "sc":{"freq":2880},"cut":{"t":1,"d":-1}}},{
    "id":17,
  "cfg":{
    "modules":{
      "EA":{"h":55}},
    "sc":{"freq":2880},"cut":{"t":1,"d":-1}}},{
    "id":18,
  "cfg":{
    "modules":{
      "EA":{"h":55}},
    "sc":{"freq":2880},"cut":{"t":1,"d":-1}}},{
    "id":20,
  "cfg":{
    "modules":{
      "EA":{"h":55}},
    "sc":{"freq":2880},"cut":{"t":1,"d":-1}}},{
    "id":25,
  "cfg":{
    "modules":{
      "EA":{"h":54}},
    "sc":{"freq":2880},"cut":{"t":1,"d":-1}}},{
    "id":26,
  "cfg":{
    "modules":{
      "EA":{"h":55}},
    "sc":{"freq":2880},"cut":{"t":1,"d":-1}}},{
    "id":27,
  "cfg":{
    "modules":{
      "EA":{"h":55}},
    "sc":{"freq":2880},"cut":{"t":1,"d":-1}}},{
    "id":28,
  "cfg":{
    "modules":{
      "EA":{"h":55}},
    "sc":{"freq":2880},"cut":{"t":1,"d":-1}}},{
    "id":29,
  "cfg":{
    "modules":{
      "EA":{"h":55}},
    "sc":{"freq":2880},"cut":{"t":1,"d":-1}}},{
    "id":31,
  "cfg":{
    "modules":{
      "EA":{"h":55}},
    "sc":{"freq":2880},"cut":{"t":1,"d":-1}}},{
    "id":33,
  "cfg":{
    "modules":{
      "EA":{"h":50}},
    "sc":{"freq":2880},"cut":{"t":1,"d":-1}}},{
    "id":34,
  "cfg":{
    "modules":{
      "EA":{"h":55}},
    "sc":{"freq":2880},"cut":{"t":1,"d":-1}}}]},"dk":{"id":[0],
    "mac":[]}},
  "dat":{"uuid":"x",
    "tm":"2024-04-15T13:20:50.955Z",
    "fw":"3.33.0+11",
    "ls":7,"le":0,"conn":"4G",
    "sc":{"slot":0},"sh":0,
    "bt":{"t":41,"v":17,"p":38,"nr":90,"c":0,"m":0},
    "dmp":[44.4363136291504,340.40051269531248,-149.40017700195312],
    "st":{"b":0,"d":0,"wt":0,"bl":0},
    "rsi":-71,"act":1,
    "rain":{"s":0,"cnt":0},
    "rtk":{"network":{"status":0,"error":0},"gps":{"status":0,"error":0},"imu":{"status":0}},
    "head":{"uuid":"x","act":1,
    "fw":"1.3.15+27"},"cut":{"z":9},
    "modules":{"US":{"stat":"ok"},
      "4G":{"mode":0,"stat":"ok","vers":"x","network":{"status":"connected","ICCID":"xxxxxx","IMSI":"x","mode":"Cat-M1","rssi":-71},"geo":{"stat":"unset"},"gps":{"coo":[x,x]}},
      "EA":{"stat":"ok","error":0}}}
}`

**Comment by MTrab** [comment]
Can't even spot the zones definition in that JSON snippet

**Comment by cm86** [comment]
Hello @MTrab,
do you still need some credentials?

Mit freundlichen Grüßen
Chris

**Comment by MTrab** [comment]
Right now it seems like I need to rethink the whole zone thing in the integration, as previously only 4 zones were supported and now there seems like no limit.

So as for now the zones support for newer devices will not work.

**Comment by cm86** [comment]
yes - that's right for the Kress RTK Mowers.

i have in the one map 9 Zones.
and in the other map 14 in the other map.
i also don't think that you have a limit in this.
the only limit i know is that the area must not exceed the maximum area of the corresponding mower.
for example the KR173E can just have a maximum area of 5000 sqm maped.

Mit freundlichen Grüßen
Chris

**Comment by cm86** [comment]
can't you look at initialisation into the json how much zones are configured.
and then pass this number into the configuration as variable.
so that the service can use the special numbers of zones?

and when the user creates a new zone and want to use this he has to reload the integration.

Mit freundlichen Grüßen 
Chris

**Comment by MTrab** [comment]
Yeah - I'll need to see how I can pass the number of zones defined to the config of the zone selection.
Also, currently the pyworxcloud only supports 4 zones, so that'll need an overhaul as well to support this structure.

**Comment by johngianni** [comment]
For now it could be sufficient to have the correct information about the current mowing zone [dat][cut][z].... I should be able to implement this is the pyworxcloud, but it's really hard for me to understand the logic of the HA integration. thank you for taking care ! :)

**Comment by MTrab** [comment]
Hving the current zone in the module wouldn't solve anything in the integration, as the integration is locked to 4 zones. So it'll cast an error if trying to set a current zone larger than number 4

**Comment by cm86** [comment]
and then kress is in the process of replacing the zone number with the zone name. at least in the app.
there will always be an id in the background. but i think that you could possibly think that far and implement it that way.
But I've just noticed that the zone names are not transmitted via mqtt. they may be stored locally on the device (phone).

Mit freundlichen Grüßen
chris

**Comment by MTrab** [comment]
If the names aren't available in the API, then there is nothing I can do as to present them in Home Assistant

**Comment by cm86** [comment]
yeah no problem - in home assistant we can work the with other things.
a template sensor would be my first shot... ;)
but that can every user make on his own - if he wan'ts.

Mit freundlichen Grüßen
Chris

**Comment by johngianni** [comment]
Zone number is more than enough !

**Comment by Piccyman** [comment]
I have the same problem, no longer reports any details other than if it's docked

**Comment by cm86** [comment]
Hello @MTrab,
are here any news/updates for this topic?

Mit freundlichen Grüßen
Chris

**Comment by MTrab** [comment]
Nothing new

**Comment by MTrab** [comment]
This should be fixed in the next release.

RTK zone reads have now been added, so the current zone and available RTK zone ids should be exposed correctly in the new codebase.

**Comment by MTrab** [comment]
Fixed in the beta release of v7
